core/utils/poly-look/src/visualisations/charts/bar-charts/horizontalBarChart/horizontalBarChart.js

```diff
@@ -0,0 +1,329 @@
+        const relevantBars = this.data.filter(
+          (data) => data.group === group.id
+        ).length;
+        const step = this._barWidth / (1 - barPaddingProportion);
+        const absolutePad = step * barPaddingProportion;
+        const range = relevantBars * step + absolutePad;
+        const scale = d3
+          .scaleBand()
+          .range([headingMargin, headingMargin + range])
+          .padding(barPaddingProportion);
+        headingMargin = headingSpace + headingMargin + range;
+        yScales.push(new ScaleContainer({ scale, id: group.id }));
+      }
+      return yScales;
+    } else {
+      return [
+        new ScaleContainer({
+          scale: d3
+            .scaleBand()
+            .range([0, this.chartHeight])
+            .padding(barPaddingProportion),
+          id: null,
+        }),
+      ];
+    }
+  }
+
+  _adaptScalesToData() {
+    this._xScale.domain([0, d3.max(this.data, (d) => d.value)]);
+    for (let scaleContainer of this._yScales) {
+      const relevantData = scaleContainer.id
+        ? this.data
+            .filter((data) => data.group === scaleContainer.id)
+            .sort((a, b) => b.value - a.value)
+            .map((data) => data.title)
+        : this.data.sort((a, b) => b.value - a.value).map((data) => data.title);
+      scaleContainer.scale.domain(relevantData);
+    }
+  }
+
+  _updateExistingBars(bars) {
+    bars
+      .transition()
+      .duration(750)
+      .attr("x", 0)
+      .attr("width", initializingBarHeight)
+      .attr("y", (d) => {
+        for (let scaleContainer of this._yScales) {
+          if (!scaleContainer.id)
+            scaleContainer.scale(d.title) +
+              scaleContainer.scale.bandwidth() / 4;
+          if (d.group === scaleContainer.id)
+            return (
+              scaleContainer.scale(d.title) +
+              (scaleContainer.scale.bandwidth() - this._barWidth) / 2
+            );
+        }
+      })
+      .attr("height", this._barWidth)
+      .attr("fill", this._barColor)
+      .attr("class", "bar")
+      .transition()
+      .duration(750)
+      .attr("x", 0)
+      .attr("width", (d) => this._xScale(d.value));
+  }
+
+  _addEnteringBars(barGroups) {
+    barGroups
+      .append("rect")
+      .attr("x", 0)
+      .attr("width", initializingBarHeight)
+      .attr("y", (d) => {
+        for (let scaleContainer of this._yScales) {
+          if (!scaleContainer.id)
+            return (
+              scaleContainer.scale(d.title) +
+              scaleContainer.scale.bandwidth() / 4
+            );
+          if (d.group === scaleContainer.id)
+            return (
+              scaleContainer.scale(d.title) +
+              (scaleContainer.scale.bandwidth() - this._barWidth) / 2
+            );
+        }
+      })
+      .attr("height", this._barWidth)
+      .attr("fill", this._barColor)
+      .attr("class", "bar")
+      .transition()
+      .duration(750)
+      .delay((_, i) => i * 50)
+      .attr("x", 0)
```

Comment:
The problem with these long stanzas is that they are tedious to type, and thus prone to errors. If there are parts of it that are repeated, possibly create a private function to DRY them.